dask/dataframe/io/io.py

```diff
@@ -750,5 +755,301 @@ def sorted_division_locations(seq, npartitions=None, chunksize=None):
+    label : str, optional
+        String to use as the function-name label in the output
+        collection-key names.
+    token : str, optional
+        String to use as the "token" in the output collection-key names.
+    enforce_metadata : bool, default True
+        Whether to enforce at runtime that the structure of the DataFrame
+        produced by ``func`` actually matches the structure of ``meta``.
+        This will rename and reorder columns for each partition,
+        and will raise an error if this doesn't work or types don't match.
+    **kwargs:
+        Key-word arguments to broadcast to each output partition. These
+        same arguments will be passed to ``func`` for every output partition.
+
+    Examples
+    --------
+    >>> import pandas as pd
+    >>> import dask.dataframe as dd
+    >>> func = lambda x, size=0: pd.Series([x] * size)
+    >>> inputs = ["A", "B"]
+    >>> dd.from_map(func, inputs, size=2).compute()
+    0    A
+    1    A
+    0    B
+    1    B
+    dtype: object
+
+    This API can also be used as an alternative to other file-based
+    IO functions, like ``read_parquet`` (which are already just
+    ``from_map`` wrapper functions):
+
+    >>> import pandas as pd
+    >>> import dask.dataframe as dd
+    >>> paths = ["0.parquet", "1.parquet", "2.parquet"]
+    >>> dd.from_map(pd.read_parquet, paths).head()  # doctest: +SKIP
+                        name
+    timestamp
+    2000-01-01 00:00:00   Laura
+    2000-01-01 00:00:01  Oliver
+    2000-01-01 00:00:02   Alice
+    2000-01-01 00:00:03  Victor
+    2000-01-01 00:00:04     Bob
+
+    Since ``from_map`` allows you to map an arbitrary function
+    to any number of iterable objects, it can be a very convenient
+    means of implementing functionality that may be missing from
+    from other DataFrame-creation methods. For example, if you
+    happen to have apriori knowledge about the number of rows
+    in each of the files in a dataset, you can generate a
+    DataFrame collection with a global RangeIndex:
+
+    >>> import pandas as pd
+    >>> import numpy as np
+    >>> import dask.dataframe as dd
+    >>> paths = ["0.parquet", "1.parquet", "2.parquet"]
+    >>> file_sizes = [86400, 86400, 86400]
+    >>> def func(path, row_offset):
+    ...     # Read parquet file and set RangeIndex offset
+    ...     df = pd.read_parquet(path)
+    ...     return df.set_index(
+    ...         pd.RangeIndex(row_offset, row_offset+len(df))
+    ...     )
+    >>> def get_ddf(paths, file_sizes):
+    ...     offsets = [0] + list(np.cumsum(file_sizes))
+    ...     return dd.from_map(
+    ...         func, paths, offsets[:-1], divisions=offsets
+    ...     )
+    >>> ddf = get_ddf(paths, file_sizes)  # doctest: +SKIP
+    >>> ddf.index  # doctest: +SKIP
+    Dask Index Structure:
+    npartitions=3
+    0         int64
+    86400       ...
+    172800      ...
+    259200      ...
+    dtype: int64
+    Dask Name: myfunc, 6 tasks
```

Comment:
@jsignell - Let me know if one or both of these other examples make sense to you.  It was hard for me to come up with a concise example that requires two `Iterable` arguments :)

Reply:
I really like this example

Reply:
Yes! Thanks so much for taking in that feedback! That example is super helpful :)